### ISSUE TITLE
Publish release

### ISSUE THREAD

**Issue by chrysn** [BennyKok/OpenLauncher#695]
Please consider tagging a version as a new release, be it 0.7.5 or 0.8.0. Such a version should then have it easy to get updated in F-Droid.

I see that a lot of bug fixes have accumulated (and experienced disappearing widgets myself) since late 2020, and in particular PR #637 (PWAs) would be a great improvement to see in a released version.

**Comment by MASTERdjango** [comment]
I have already compiled it, the only thing that has failed has been the backups....
seems it can not find the zip file....
or create it...

**Comment by MASTERdjango** [comment]
there you go: https://github.com/MASTERdjango/openLAUNCHER, It just works... and yes PR (https://github.com/OpenLauncherTeam/openlauncher/pull/637) (PWAs) works... expect bugs... I just built the APK today.

**Comment by gsantner** [comment]
@MASTERdjango Everything merged on master is available from CI build apk. No need to upload some apk for somebody.

**Comment by hamishmb** [comment]
Are you still in need of beta testers? I'm happy to try a dev build as well, if that would help. The current stable release seems pretty buggy.

**Comment by nazar-pc** [comment]
I started testing debug build from CI from 4abc6d27eaad593ca1fa664d8f36db4db2b8f7a9 too, seems to be working just fine so far, in fact better than latest release on F-Droid.

I'd say make it a preview/beta release on F-Droid, it does have such a feature. Those who need stable version can continue to use stable release, those who want to test preview can opt-in in F-Droid settings.

**Comment by nazar-pc** [comment]
Used for over a week, no issues whatsoever while the published version was crashing for me on Android 13 (I exported and imported the same settings).

**Comment by hamishmb** [comment]
Crashes a lot on Android 11 for me too.

**Comment by User2564** [comment]
@MASTERdjango and @gsantner,
It would be greatly appreciated if I could have some assistance in setting up a functional build environment (ideally running within a Debian base WSL 1 running on Windows 10).

A previous example using a generic online guide did not go down very well to say the least.

It might be impractical but it would be appreciated if Artifact expiration could be disabled.

**Comment by Ptitg** [comment]
For my use, I find the master version better than 7.4. I use it daily.

**Comment by User2564** [comment]
Via working with ChatGPT I have been able to setup my own build environment based around Debain running within Windows Subsystem for Linux (WSL) version I.

Prompt
------------------
Good evening,
Can you please provide a detailed build process for the Android Application called "Open Launcher".
Desired environment: A Debian 9 install stored within WSL 1 and Windows 10 as the host

The source code can be found at https://github.com/OpenLauncherTeam/openlauncher

Output
-------------------
Sure, here are the steps to build the Open Launcher Android application on a Debian 9 install within WSL 1 and Windows 10 as the host:

Install necessary tools:

Java Development Kit (JDK) version 8 or above
Android Software Development Kit (SDK)
Gradle build tool
Git
Clone the Open Launcher repository:

Open a terminal window and navigate to the directory where you'd like to store the project.
Run the command $ git clone https://github.com/OpenLauncherTeam/openlauncher.git to clone the repository.
Set up environment variables:

Set JAVA_HOME to the path of your installed JDK.
Set ANDROID_HOME to the path of your installed Android SDK.
Install Android SDK components:

Run the command $ sdkmanager "platforms;android-29" "build-tools;29.0.2" to install the necessary SDK components.
Build the project:

Navigate to the root directory of the project.
Run the command $ ./gradlew assembleDebug to build the project in debug mode.
Alternatively, run the command $ ./gradlew assembleRelease to build the project in release mode.
Find the APK:

The APK file will be generated in openlauncher/app/build/outputs/apk/debug or openlauncher/app/build/outputs/apk/release, depending on the build mode selected.
That's it! You now have a built version of the Open Launcher Android application.

-------------------

To be more useful:

sudo apt-get install openjdk-8-jre

To use sdkmanager (https://developer.android.com/tools/sdkmanager), I then downloaded the Linux port of ther SDK-platform tools (https://developer.android.com/tools/releases/platform-tools).

After tripple reading the instructions and using some french, I came up with the following file struture.

PATH/android_sdk
 - cmdline-tools
  - latest
   - Bin
   - lib
 - platform-tools
 ...............

As to be expected I then had to update the debian image at some point.

Within the src folder I executed the following commands
export ANDROID_SDK_ROOT=PATH/android_sdk/ e.g. export ANDROID_SDK_ROOT=/mnt/c/Users/03520/Desktop/android_sdk/

cd /mnt/c/Users/03520/Desktop/android_sdk/
./../android_sdk/cmdline-tools/latest/bin/sdkmanager "platforms;android-29" "build-tools;29.0.2"

I have the following shell script to help automate building my own custom release which sits next to the src folder and I call from within the src folder.
--------------------------------------

export ANDROID_SDK_ROOT=/mnt/c/Users/03520/Desktop/android_sdk/
git merge --no-commit veryMinorTweaks # As time goes though I intended to switch over to git merge --no-commit $(git branch --list | grep -v "* master" | paste -s)
./gradlew assembleDebug
git reset --hard HEAD -q
cp app/build/outputs/apk/flavorDefault/debug/app-flavorDefault-debug.apk ../OpenLauncher-modified-$(date +"%Y-%m-%d-%H.%M.%S")-Upstream-Master-commit-8a7afbe-Default-unsigned-debug.apk

**Comment by erebion** [comment]
There was a release on F-Droid a month ago. \o/

**Comment by nazar-pc** [comment]
Yes, release is out on F-Droid, but for some reason it is created for ancient devices (Android 4.2+) and as the result doesn't support auto-updates. Would be great to have release for modern versions of Android.

**Comment by erebion** [comment]
> Would be great to have release for modern versions of Android.

Well, that would break it for people running ancient versions of Android. Maybe then there should also be a version that is built for older Android for a while, as long as it does not get complicated.

**Comment by nazar-pc** [comment]
No one should be using ancient versions of Android at this point, they're horribly insecure.

But those who do can still install older version and continue using it, I doubt they expect latest and greatest 12 years later.

**Comment by erebion** [comment]
Much of the world is poor and lacks access.

**Comment by nazar-pc** [comment]
Access to older versions of the software?

**Comment by gsantner** [comment]
OpenLauncher is not developed anymore. You might use something else.

There was a final release before that. Older versions can be found on F-Droid.

**Comment by nazar-pc** [comment]
@gsantner it would be great if you can trigger CI again and upload apk to GitHub release. F-Droid removed the app from their store and GitHub pruned Actions artifacts, so there is no way to download the app anymore.

**Comment by gsantner** [comment]
You should find it on F-Droid Archive. Expect nothing to happen anymore.

**Comment by nazar-pc** [comment]
As I said they have removed it from F-Droid.

Here is 1.0 release apk I extracted from my phone in case someone needs is: [open-launcher.zip](https://github.com/user-attachments/files/18133566/open-launcher.zip)

**Comment by JonnyTech** [comment]
> You should find it on F-Droid Archive.

In the fdroid app, select the repository then scroll to the end and enable the archive:

![Screenshot_20241214-022448](https://github.com/user-attachments/assets/05ebb702-05d3-46d3-bc97-f9e2b84e5189)

Then refresh the apps list and search for openlauncher:

![Screenshot_20241214-022630](https://github.com/user-attachments/assets/3853efe7-596b-49eb-a1ab-faad53913185)

**Comment by JonnyTech** [comment]
Direct link: https://f-droid.org/archive/com.benny.openlauncher_100.apk
